### ISSUE TITLE
plugin.configs is undefined in esm mode

### ISSUE THREAD

**Issue by crfrolik** [typescript-eslint/typescript-eslint#8675]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-eslint

### Repro Code

```TypeScript
1. Install `typescript-eslint` at 7.2.0
2. Use ESM mode (`"type":"module"` in `package.json`)
3. Create `eslint.config.js`

Add:


import { plugin } from 'typescript-eslint';

console.log(plugin.configs);
```

This works fine in CommonJS mode.

Note that I am trying to access `plugin.configs.recommended`, but even the parent object `configs` isn't there.
```


### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/<rule-name>": ["error", ...<options>],
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

I expect the plugin to have `plugin.configs.recommended`, but this does not exist.`

### Actual Result

`configs` is `undefined`

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `typescript-eslint`     | `7.2.0` |
| `TypeScript`                           | `5.3.3` |
| `ESLint`                               | `8.57.0` |
| `node`                                 | `20.11.1` |

**Comment by bradzacher** [comment]
https://typescript-eslint.io/getting-started

Check out our getting started guide and you'll see what you did wrong. The configs are a top level export.
